src/main/java/org/spongepowered/api/GameRegistry.java

```diff
@@ -296,6 +300,19 @@
     void registerWorldGeneratorModifier(WorldGeneratorModifier modifier);
 
     /**
+     * Registers a new {@link AbstractAITask} with an {@link Agent} as the owner. The complete id
+     * will be in the format of <code>{@link PluginContainer#getId()}:id</code>.
+     *
+     * @param plugin The plugin who owns it
+     * @param id The id that represents the task type
+     * @param ownerClass The class of the owner
+     * @param aiClass The class of the task
+     * @return The type
+     */
+    AITaskType registerAITaskType(Object plugin, String id, String name, Class<? extends Agent> ownerClass, Class<? extends AbstractAITask<? extends
```

Comment:
Should `ownerClass` be the same as the class in the `AbstractAITask` generic?
If so then should use a generic on the method `<T extends Agent>` ... `Class<T> ownerClass, Class<? extends AbstractAITask<T>>`